server/routerlicious/packages/routerlicious-base/src/utils/sessionHelper.ts

```diff
@@ -134,16 +135,32 @@ async function updateExistingSession(
         isSessionActive: false,
     };
     try {
-        await documentsCollection.upsert(
+        const result = await documentsCollection.findOrCreate(
             {
                 documentId,
+                "session.isSessionAlive": false,
             },
             {
+                createTime: document.createTime,
                 deli: updatedDeli ?? document.deli,
-                scribe: updatedScribe ?? document.scribe,
+                documentId: document.documentId,
                 session: updatedSession,
-            },
-            null);
+                scribe: updatedScribe ?? document.scribe,
+                tenantId: document.tenantId,
+                version: document.version,
+            });
+        // There is no document with isSessionAlive as false. It means this session has been discovered by
```

Comment:
I am not sure if this is the only case where result.existing is false. For instance, if the document with this docid was got deleted for any reason, I would assume that it would still return false.

Reply:
If the document with this doc is deleted, we cannot find the document with id before hit this method. The existingSession variable would be false, and we would hit createNewSession method.